server/fractal.js

```diff
@@ -21,9 +21,50 @@ fractal.web.set('builder.dest', './../cardigan');
 /* Set the default status of components */
 fractal.components.set('default.status', 'wip');
 
-/* Add paths for component templates */
+/* Add extensions */
 const nunj = require('@frctl/nunjucks')({
-    //paths: ['../dist/assets/js/inline/']
+    extensions: {
+      componentExtension: (function () {
+
+        function ComponentExtension() {
+          this.tags = ['component'];
+
+          this.parse = function(parser, nodes, lexer) {
+            const token = parser.nextToken();
+
+            const args = parser.parseSignature(null, true);
+            parser.advanceAfterBlockEnd(token.value);
+
+            const body = parser.parseUntilBlocks('endcomponent');
+            parser.advanceAfterBlockEnd();
+
+            return new nodes.CallExtensionAsync(this, 'run', args);
+          };
+
+          this.run = function() {
+              const source = fractal.components;
+              const args = Array.from(arguments);
+              const callback = args.pop();
+              args.shift();
+              const handle = args[0];
+              let context = args[1];
+              const entity = source.find('@'+handle);
+              
+              if (!entity) {
+                  throw new Error(`Could not render component '${handle}' - component not found.`);
+              }
+
+              entity.render(context).then(html => {
+                  callback(null, html);
+              });
+          };
+        };
+
+        return new ComponentExtension();
+
+      })()
```

Comment:
Is there a reason we don't just import the Component from the one we use on the server app?
We'll land up with two versions like this?